File: java/src/main/java/com/penapereira/patterns/decorator/Decorator.java
package com.penapereira.patterns.decorator;

/** Holds the wrapped component and forwards to it; subclasses add behaviour. */
public abstract class Decorator implements Component {

    private final Component component;

    protected Decorator(Component component) {
        this.component = component;
    }

    @Override
    public String operation() {
        return component.operation();
    }
}

File: python/src/patterns/decorator/decorator.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .component import Component


class Decorator:
    """Holds the wrapped component and forwards to it; subclasses add behaviour."""

    def __init__(self, component: Component) -> None:
        self._component = component

    def operation(self) -> str:
        return self._component.operation()
